mava/wrappers/env_preprocess_wrappers.py

```diff
@@ -71,12 +75,16 @@ def reset(self) -> dm_env.TimeStep:
             dm_env.TimeStep(
                 timestep.step_type, timestep.reward, timestep.discount, new_observations
             ),
-            extras,
+            env_extras,
         )
 
     def step(self, actions: Dict) -> dm_env.TimeStep:
         """Step the environment and concat agent ID"""
-        timestep, extras = self._environment.step(actions)
+        timestep = self._environment.step(actions)
```

Comment:
What changed that resulted in the need to add this code?

Reply:
Certain environments mainly the debugging ones only return a timestep without any extras. So when I was testing on debugging environments, I realised we need to slight modify the code this way so that it handle such environments. Maybe if we delete all such environments we will not need this anymore

Reply:
I see. Fair enough :)

Reply:
A followup question. Why is extras needed now when it was not in the past for this environment?

Reply:
Another way around this is to make the environment return an empty dictionary as its extras.

Reply:
Actually I just saw that in different areas in Mava, this trick is applied. So maybe I think it wasn't include only in these wrappers. You can see the same [here](https://github.com/instadeepai/Mava/blob/develop/mava/environment_loop.py#L120). I prefer we always do this rather than tracking down every environment.

@DriesSmit regarding your question. The way the wrappers are implemented is that their step function calls the step function of main environemnt. Then modifies the output before returning it to the executor. The current implementation assumed every wrapper returns a timestep and an extras dict. I only modified this to make it capable of wrapping the two different kinds of environments currently supported in Mava.

Reply:
Yes I code it up long ago to handle both types of environments. Okay it seem fine then :slightly_smiling_face: Thanks @ulricharmel. @RuanJohn what do you think?

Reply:
This looks good to me in that case. Thank you @ulricharmel :)

Reply:
Is it possible to move this fix to its own PR? That way we can get it in and the PettingZoo agent ID fix will lead to IDs that can be concatentated.